### PR TITLE
feat(insight): create insight client

### PR DESCRIPTION
**Pull request by aelfaleh**
https://coveord.atlassian.net/browse/SVCC-2246

### CODE REVIEW

File: src/insight/insightClient.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+export interface InsightClientOptions extends ClientOptions {
+    enableAnalytics: boolean;
+}
+
+export class CoveoInsightClient {
+    public coveoAnalyticsClient: AnalyticsClient;
+
+    constructor(private opts: Partial<InsightClientOptions>, private provider: InsightClientProvider) {
+        this.coveoAnalyticsClient =
+            opts.enableAnalytics === false ? new NoopAnalytics() : new CoveoAnalyticsClient(opts);
```
Comment (lbergeron) [inline src/insight/insightClient.ts:29]: We'll need to handle the `doNotTrack` flag like in this PR.
https://github.com/coveo/coveo.analytics.js/pull/349

**Comment by nathanlb** [comment]
FYI this other PR is likely relevant to this change https://github.com/coveo/coveo.analytics.js/pull/349

File: src/insight/insightClient.ts (hunk 1 above)
Comment (aelfaleh) [reply src/insight/insightClient.ts:29]: I'll wait for the merge , to make changes

**Review by lbergeron — COMMENTED** [review]
I think only the doNotTrack is missing, everything else look all good to me.

**Review by nathanlb — APPROVED** [review]
LGTM

**Pull request merged**